ADAL/src/validation/ADAuthorityValidation.m

```diff
@@ -137,8 +137,9 @@ - (BOOL)isAuthorityValidated:(NSURL *)authority domain:(NSString *)domain
 
 #pragma mark - Authority validation
 
-- (void)validateAuthority:(ADRequestParameters*)requestParams
-          completionBlock:(ADAuthorityValidationCallback)completionBlock
+- (void)checkAuthority:(ADRequestParameters*)requestParams
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function has 4 returns, which is more than the 3 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1142)

Reply:
Can we get this rule removed? I'd much rather have people split out if error checks to make the code more readable then have to deal with an arbitrary # of returns requirement.

Reply:
^^ I agree